src/plugins/intel_cpu/src/nodes/kernels/registers_pool.hpp

```diff
@@ -0,0 +1,305 @@
+            }
+            if (!isFreeIndexVector[regIdx]) {
+                IE_THROW() << "Inconsistency in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            isFreeIndexVector[regIdx] = false;
+        }
+
+        void setAsUnused(int regIdx) {
+            if (regIdx >= isFreeIndexVector.size() || regIdx < 0) {
+                IE_THROW() << "regIdx is out of bounds in RegistersPool::PhysicalSet::setAsUsed()";
+            }
+            if (isFreeIndexVector[regIdx]) {
+                IE_THROW() << "Inconsistency in RegistersPool::PhysicalSet::setAsUnused()";
+            }
+            isFreeIndexVector[regIdx] = true;
+        }
+
+        int getUnused(int requestedIdx) {
+            if (requestedIdx == anyIdx) {
+                return getFirstFreeIndex();
+            } else {
+                if (requestedIdx >= isFreeIndexVector.size() || requestedIdx < 0) {
+                    IE_THROW() << "requestedIdx is out of bounds in RegistersPool::PhysicalSet::getUnused()";
+                }
+                if (!isFreeIndexVector[requestedIdx]) {
+                    IE_THROW() << "The register with index #" << requestedIdx << " already used in the RegistersPool";
+                }
+                return requestedIdx;
+            }
+        }
+
+        void exclude(Xbyak::Reg reg) {
+            isFreeIndexVector.at(reg.getIdx()) = false;
+        }
+
+        size_t countUnused() const {
+            size_t count = 0;
+            for (const auto& isFree : isFreeIndexVector) {
+                if (isFree) {
+                    ++count;
+                }
+            }
+            return count;
+        }
+
+    private:
+        int getFirstFreeIndex() {
+            for (int c = 0; c < isFreeIndexVector.size(); ++c) {
+                if (isFreeIndexVector[c]) {
+                    return c;
+                }
+            }
+            IE_THROW() << "Not enough registers in the RegistersPool";
+        }
+
+    private:
+        std::vector<bool> isFreeIndexVector;
+    };
+
+    virtual int getFreeOpmask(int requestedIdx) { IE_THROW() << "getFreeOpmask: The Opmask is not supported in current instruction set"; }
+    virtual void returnOpmaskToPool(int idx) { IE_THROW() << "returnOpmaskToPool: The Opmask is not supported in current instruction set"; }
+    virtual size_t countUnusedOpmask() const { IE_THROW() << "countUnusedOpmask: The Opmask is not supported in current instruction set"; }
+
+    RegistersPool(std::initializer_list<Xbyak::Reg> regsToExclude, int simdRegistersNumber)
+            : simdSet(simdRegistersNumber) {
+        checkUniqueAndUpdate();
+        for (auto& reg : regsToExclude) {
+            if (reg.isXMM() || reg.isYMM() || reg.isZMM()) {
+                simdSet.exclude(reg);
+            } else if (reg.isREG()) {
+                generalSet.exclude(reg);
+            }
+        }
+        generalSet.exclude(Xbyak::Reg64(Xbyak::Operand::RSP));
```

Comment:
Also I suggest to exclude RAX, RCX, RDI. They are used by Xbyak and can be cause of difficult debugged issues.

Reply:
@nshchego  I added the default constructor for the IsaRegistersPool class, which excludes some of these critical registers by default.